beacon-chain/core/state/state.go

```diff
@@ -51,10 +52,54 @@ import (
 //	    state.active_index_roots[index] = active_index_root
 //	    state.compact_committees_roots[index] = committee_root
 //	  return state
+// This method differs from the spec so as to process deposits beforehand instead of the end of the function.
 func GenesisBeaconState(deposits []*ethpb.Deposit, genesisTime uint64, eth1Data *ethpb.Eth1Data) (*pb.BeaconState, error) {
```

Comment:
Is there a plan to deprecate this in the future or maybe delete this? It's only really used in tests and don't like the pattern of having duplicated functions with slightly different functionality

Reply:
See #4446